src/Illuminate/Foundation/Auth/SendsPasswordResetEmails.php

```diff
@@ -82,6 +86,12 @@ protected function sendResetLinkResponse(Request $request, $response)
      */
     protected function sendResetLinkFailedResponse(Request $request, $response)
     {
+        if ($request->wantsJson()) {
```

Comment:
@taylorotwell This method could just throw the `ValidationException`, the condition is not needed because the response is handled in the exception handler: https://github.com/laravel/framework/blob/6.x/src/Illuminate/Foundation/Exceptions/Handler.php#L258

Reply:
@sebdesign I think it's to avoid returning the redirect response from the `back` helper below

Reply:
@matt-allan if the exception is thrown the code won't reach the return statement. Maybe it's just for the sake of honoring the return docblock?

Reply:
Note the `withInput` on line 96. That won't work if I just throw a ValidationException.